src/jit/codegenlinear.cpp

```diff
@@ -1075,6 +1022,8 @@ void CodeGen::genCheckConsumeNode(GenTree* treeNode)
 //    of the multi-reg return.
 regNumber CodeGen::genConsumeReg(GenTree* tree)
 {
+    assert(!tree->isContained());
```

Comment:
I see that the bug was actually an assert, so indeed it would cause SBCG in a release build, but was caught by an assert.
I think that the correct approach is to fix the bug where the caller was not checking for containedness, and then separately propose a change in abstraction.
Clearly the proposed abstraction requires further discussion.

---
In reply to: [85578253](https://github.com/dotnet/coreclr/pull/7876#discussion_r85578253) [](ancestors = 85578253)